atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/RowColumnRangeExtractor.java

```diff
@@ -80,32 +83,46 @@ public void extractResults(Iterable<byte[]> canonicalRows,
             byte[] rawRow = CassandraKeyValueServices.getBytesFromByteBuffer(colEntry.getKey());
             byte[] row = canonicalRowsByHash.get(Sha256Hash.computeHash(rawRow));
             List<ColumnOrSuperColumn> columns = colEntry.getValue();
-            if (!columns.isEmpty()) {
-                rowsToLastCompositeColumns.put(row, columns.get(columns.size() - 1).column);
-            } else {
-                emptyRows.add(row);
-            }
-            rowsToRawColumnCount.put(row, columns.size());
-            for (ColumnOrSuperColumn c : colEntry.getValue()) {
-                Pair<byte[], Long> pair = CassandraKeyValueServices.decomposeName(c.column);
-                internalExtractResult(startTs, row, pair.lhSide, c.column.getValue(), pair.rhSide);
+
+            updateAppropriateDataStructures(row, columns);
+
+            for (ColumnOrSuperColumn c : columns) {
+                Pair<byte[], Long> pair = CassandraKeyValueServices.decomposeName(c.getColumn());
+                internalExtractResult(startTs, row, pair.lhSide, c.getColumn().getValue(), pair.rhSide);
             }
         }
     }
 
+    private void updateAppropriateDataStructures(byte[] row, List<ColumnOrSuperColumn> columns) {
+        if (!columns.isEmpty()) {
+            rowsToLastCompositeColumns.put(row, columns.get(columns.size() - 1).getColumn());
+        } else {
+            emptyRows.add(row);
+        }
+        rowsToRawColumnCount.put(row, columns.size());
+    }
+
     private void internalExtractResult(long startTs, byte[] row, byte[] col, byte[] val, long ts) {
         if (ts < startTs) {
             Cell cell = Cell.create(row, col);
             if (!collector.containsKey(row)) {
-                collector.put(row, new LinkedHashMap<Cell, Value>());
+                collector.put(row, new LinkedHashMap<>());
                 collector.get(row).put(cell, Value.create(val, ts));
             } else if (!collector.get(row).containsKey(cell)) {
                 collector.get(row).put(cell, Value.create(val, ts));
             }
+        } else {
+            getPostFilteredCellsMeter().mark();
         }
     }
 
     public RowColumnRangeResult getRowColumnRangeResult() {
         return new RowColumnRangeResult(collector, rowsToLastCompositeColumns, emptyRows, rowsToRawColumnCount);
     }
+
+    private Meter getPostFilteredCellsMeter() {
+        // TODO(hsaraogi): add table names as a tag
+        return metricsManager.registerMeter(RowColumnRangeExtractor.class, "startTsCellFilterCount");
```

Comment:
might want to put this into an instance variable to avoid the overhead of looking it up on every call

Reply:
Done.